internal/serverinstall/nomad.go

```diff
@@ -813,29 +853,61 @@ func waypointNomadJob(c nomadConfig, rawRunFlags []string) *api.Job {
 	grpcPort, _ := strconv.Atoi(defaultGrpcPort)
 	httpPort, _ := strconv.Atoi(defaultHttpPort)
 
-	// Include services to be registered in Consul. Currently configured to happen by default
+	// Include services to be registered. Currently configured to happen with Consul by default
 	// One service added for Waypoint UI, and one for Waypoint backend port
-	if c.consulService {
+	var services []*api.Service
+	if (c.serviceProvider == "consul" && (c.consulService || upgrade)) || (c.consulService && (c.serviceProvider == "consul" || upgrade)) {
 		token := ""
 		if c.consulToken == "" {
 			token = os.Getenv("CONSUL_HTTP_TOKEN")
 		} else {
 			token = c.consulToken
 		}
 		job.ConsulToken = &token
```

Comment:
```suggestion
		token := c.consulToken
		if token == "" {
			token = os.Getenv("CONSUL_HTTP_TOKEN")
		}
		job.ConsulToken = &token

```

I'm not sure if we should check if token is empty before storing it in `job.ConsulToken` or if it's fine to put something empty in there ¯\\_(ツ)_/¯